### DISCUSSION TITLE
Required if published = true -- Validation Rule on Blueprint Fields

### DISCUSSION THREAD

**Discussion by bradfx** [statamic/cms#6250] [Troubleshooting] [answered]
**Problem:** 
Is it possible to set fields `required` only if the entry is `published`?

I've tried `required_if:published,true` which in theory should work because the `published` value on the collection array does output `true` both in antlers templates and in the PHP validation objects. But I can't make this validation work.

I've also tried the opposite, expecting `published` to equal `false` which should be even more accurate because the entry's MD front matter literally says `published: false` in it, yet still the validation neither passes nor fails, it just does nothing.

**Reasoning:** 
I wish to save draft incomplete entries without having to enter "TODO" in every required field until I have the data I need. Progressively building up our entries would be really helpful. It sucks that we can't save anything until we have everything. I would assume for most people the required fields aren't actually required until it's published right? So how does one progressively create a draft entry without having TODO written everywhere?

**Comment by bradfx** [comment, marked as answer]
Here's what I'm trying but can't get to work.

![Screen Shot 2022-06-27 at 13 32 03](https://user-images.githubusercontent.com/1085976/175854886-975474fd-030a-4eb3-8d2e-3423f1d1b0b9.png)

**Reply by ryanmitchell** [reply]
Yeah it seems that the published field isnt passed to the validator. Looks like a bug to me.

**Reply by jesseleite** [reply]
Haven't tested this, but does `required_if:status,published` work?

**Reply by jesseleite** [reply]
^ Ignore this, it does not 😁

**Reply by jesseleite** [reply]
@bradleyflood Feel free to follow this PR here: https://github.com/statamic/cms/pull/6353
